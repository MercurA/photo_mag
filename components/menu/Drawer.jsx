import styles from './menu.module.css'

const LIST = ['Photos', 'Shop']

const Drawer = ({ isOpen, handleDrawer }) => {
    return (
        <div className={`${styles.drawer} ${isOpen ? styles.drawerOpen : styles.drawerClosed}`}>
            <div className={styles.drawerHeader}>
                <div className={styles.drawerCloseBtn} onClick={handleDrawer}>x</div>
            </div>
            <div className={styles.drawerBody}>
                {LIST.map((el, i) => (
                    <div key={i} className={styles.drawerItem}>{el}</div>
                ))}
            </div>
        </div>
    )
}

export default Drawer