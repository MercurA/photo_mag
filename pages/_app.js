import '../styles/globals.css'
import { createContext, useReducer } from "react"
import reducer, { initialState } from "../utils/state/reducer"

export const AppContext = createContext(null)

const MyApp = ({ Component, pageProps }) => {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={{ ...state, dispatch }}>
      <Component {...pageProps} />
    </AppContext.Provider>
  )
}

export default MyApp