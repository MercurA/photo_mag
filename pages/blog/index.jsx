import DesktopBlogContainer from '../../containers/DesktopBlogContainer'

const Blog = () => {
    return (
        <DesktopBlogContainer />
    )
}

export default Blog;