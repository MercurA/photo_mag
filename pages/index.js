import { createContext, useReducer } from "react";
import CardContainer from "../components/card_image/CardContainer";
import Menu from "../components/menu/Menu";
import MusicContainer from "../components/music/MusicContainer";
import reducer, { initialState } from "../utils/state/reducer";

export const AppContext = createContext(null)

export default function Home() {
  const [state, dispatch] = useReducer(reducer, initialState)

  return (
    <AppContext.Provider value={state}>
      <div >
        <Menu dispatch={dispatch}/>
        <CardContainer />
        <MusicContainer dispatch={dispatch}/>
      </div>
    </AppContext.Provider>
  )
}
