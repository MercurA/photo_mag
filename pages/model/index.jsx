import { useRef, useState } from 'react'
import styles from './model.module.css'
import { Canvas, useFrame } from '@react-three/fiber'

const Sphere = ({position, size, color}) => {
    const [hovered, setHovered] = useState(false)
    const sphereRef = useRef()
    useFrame((state, delta) => {
        sphereRef.current.rotation.y += delta/6
    })
    return (
        <mesh ref={sphereRef} onPointerOver={(e) => {e.stopPropagation(); setHovered(!hovered)}}>
            <sphereGeometry />
            <meshStandardMaterial color={hovered ? color : 'lightblue'} wireframe/>
        </mesh>
    )
}

const Model = () => {
    return (
        <div className={styles.container}>
            <Canvas >
                <directionalLight position={[0, 0, 7]} />
                <Sphere position={[0,0,0]} size={[1,1,1]} color={'orange'}/>
            </Canvas>
        </div>
    )
}

export default Model