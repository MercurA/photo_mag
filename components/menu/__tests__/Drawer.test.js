import { render } from '@testing-library/react'
import { AppContext } from '../../../pages'
import Drawer from '../Drawer'

describe('Menu', () => {
    it('should render corectly', () => {
        const state = {
            isDrawerOpen: false,
            isPopupOn: false
        }
        const comp = render(
            <AppContext.Provider value={state}>
                <Drawer />
            </ AppContext.Provider>
        )

        expect(comp).toMatchSnapshot()
    })
})