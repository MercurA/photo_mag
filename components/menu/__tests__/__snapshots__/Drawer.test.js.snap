// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render corectly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="drawer drawerClosed"
      >
        <div
          class="drawerHeader"
        >
          <button
            class="drawerCloseBtn"
          >
            <img
              alt="close icon"
              data-nimg="1"
              decoding="async"
              height="15"
              loading="lazy"
              src="/icons/close.svg"
              style="color: transparent;"
              width="15"
            />
          </button>
        </div>
        <div
          class="drawerBody"
        >
          <div
            class="drawerItem"
            data-id="black&white"
          >
            Black & White
          </div>
          <div
            class="drawerItem"
            data-id="landscape"
          >
            Landscape
          </div>
          <div
            class="drawerItem"
            data-id="people&cities"
          >
            People and cities
          </div>
        </div>
        <div
          class="drawerFooter"
        >
          <p>
            constandacheh@gmail.com
          </p>
          <div>
            <a
              class="link"
              href="https://www.instagram.com/cypher_mercurius/"
              target="_blank"
            >
              <img
                alt="instagram logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/insta_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://github.com/MercurA"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/icons/github_white.svg"
                style="color: transparent;"
                width="20"
              />
            </a>
            <a
              class="link"
              href="https://soundcloud.com/dionis5"
              target="_blank"
            >
              <img
                alt="github logo"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
                srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="drawer drawerClosed"
    >
      <div
        class="drawerHeader"
      >
        <button
          class="drawerCloseBtn"
        >
          <img
            alt="close icon"
            data-nimg="1"
            decoding="async"
            height="15"
            loading="lazy"
            src="/icons/close.svg"
            style="color: transparent;"
            width="15"
          />
        </button>
      </div>
      <div
        class="drawerBody"
      >
        <div
          class="drawerItem"
          data-id="black&white"
        >
          Black & White
        </div>
        <div
          class="drawerItem"
          data-id="landscape"
        >
          Landscape
        </div>
        <div
          class="drawerItem"
          data-id="people&cities"
        >
          People and cities
        </div>
      </div>
      <div
        class="drawerFooter"
      >
        <p>
          constandacheh@gmail.com
        </p>
        <div>
          <a
            class="link"
            href="https://www.instagram.com/cypher_mercurius/"
            target="_blank"
          >
            <img
              alt="instagram logo"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/icons/insta_white.svg"
              style="color: transparent;"
              width="20"
            />
          </a>
          <a
            class="link"
            href="https://github.com/MercurA"
            target="_blank"
          >
            <img
              alt="github logo"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/icons/github_white.svg"
              style="color: transparent;"
              width="20"
            />
          </a>
          <a
            class="link"
            href="https://soundcloud.com/dionis5"
            target="_blank"
          >
            <img
              alt="github logo"
              data-nimg="1"
              decoding="async"
              height="20"
              loading="lazy"
              src="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75"
              srcset="/_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fsoundcloud.jpg&w=48&q=75 2x"
              style="color: transparent;"
              width="20"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
